Lib/threading.py

```diff
@@ -924,6 +929,8 @@ def _bootstrap_inner(self):
             self._set_tstate_lock()
             if _HAVE_THREAD_NATIVE_ID:
                 self._set_native_id()
+            if _HAVE_SET_THREAD_NAME:
+                _set_thread_name(self.name)
```

Comment:
This should probably be more sophisticated, e.g. `f"{sys.executable} [thread {self.name!r}]"`